src/activities/menu/ConfigurationItem.qml

```diff
@@ -404,6 +413,56 @@ Item {
                 }
             }
         }
+        
+        Flow {
+            id: backgroundMusicRow
+            width: parent.width
+            spacing: 5 * ApplicationInfo.ratio
+            
+            property bool haveBackgroundMusicResource: false
+
+            function backgroundMusicChanged() {
+                backgroundMusicRow.haveBackgroundMusicResource = DownloadManager.isDataRegistered(
+                            "backgroundMusic-" + ApplicationInfo.CompressedAudio + ".rcc" )
+                         }
+
+
+            Connections {
+                target: DownloadManager
+                onDownloadFinished: backgroundMusicRow.backgroundMusicChanged()
+            }
+            
+            
+            GCText {
+                id: backgroundMusicText
+                text: qsTr("Background music")
+                fontSize: mediumSize
+                wrapMode: Text.WordWrap
+            }
+            
+            Image {
+                id: backgroundMusicImage
+                sourceSize.height: 30 * ApplicationInfo.ratio
+                source: backgroundMusicRow.haveBackgroundMusicResource ? "qrc:/gcompris/src/core/resource/apply.svg" :
+                "qrc:/gcompris/src/core/resource/cancel.svg"
+            }
+            
+            Button {
+                id: backgroundMusicButton
+                height: 30 * ApplicationInfo.ratio
+                visible: ApplicationInfo.isDownloadAllowed
+                text: backgroundMusicRow ? qsTr("Check for updates") :
+                qsTr("Download")
+                style: GCButtonStyle {}
+                onClicked: {
+                if(DownloadManager.downloadResource(DownloadManager.getBackgroundMusicResources())) {
```

Comment:
there, you download without checking if it was already present. If it is already present, you should just enable the sounds. Anyway, this button will disappear, it will be the code for the enableBackgroundMusicBox GCCheckBox.